docs/HowToDevelopThingsUsingSTDK.md

```diff
@@ -0,0 +1,256 @@
+# How to develop things using STDK based TizenRT   
+
+STDK is one of Develop Kit(STDK) for Smart Things Eco-system.   
+This guide shows an example how to enable STDK and develop things.   
+
+So now, it will explain how to develop Led Things of SmartThings and Control it through SmartThings Mobile app.   
+
+## Prerequisites   
+The goal of this guide is to execute a sample application on esp32 board. so this doc will describe an experience how to build and run a STDK App. For this example, It need a ESP-WROVER-KIT board.   
+    
+* Install packages.   
+  Below package are needed to build.   
+```shell
+  sudo apt-get install gcc git wget make libncurses-dev flex bison gperf \
+      python python-pip python-setuptools \
+      python-serial python-cryptography python-future python-pyparsing python-pyelftools 
+
+  pip install pyserial
+```
+* Board specific prerequisites [[Details]](../README.md#supported-board--emulator).   
+* Toolchain installation [[Details]](../README.md#environment-setup).   
+
+### Get TizenRT Source.   
+#### Clone TizenRT source code.   
+```shell
+git clone https://github.com/Samsung/TizenRT.git
+cd TizenRT
+TIZENRT_BASEDIR="$PWD"
+```
+#### Apply patch for STDK.   
++ Update submodule   
+```shell
+git submodule update --init "external/stdk/st-device-sdk-c"
+git submodule update --init "external/libsodium/libsodium"
+```
++ Apply patch for stdk   
+```shell
+cd external/stdk/st-device-sdk-c
+git apply ../patch/0001-port-net-fix-ssl-issue-on-TizenRT.patch
+```
+
+## Create device identity   
+In order for your IoT device to connect to the SmartThings Cloud, there are required authentication data files. To create these authentication data, you must know your MNID in the personal information window of [Developer Workspace](https://smartthings.developer.samsung.com/). Please sign in with Samsung Account and then check your MNID first.   
+
+Open a terminal window and run the following stdk-keygen command to create a Device Identity.   
+
+```bash
+$ cd external/stdk/st-device-sdk-c/tools/keygen/linux
+$ ./stdk-keygen -h
+usage: stdk-keygen -m MNID [-f firmware_version]
+
+$ ./stdk-keygen -m **** -f V201910       # replace '****' with your MNID
+Go Device Identity of Developer Workspace.
+
+Serial Number:
+┌──────────┬──────────┐
+│ STDK**** │ ce**2**3 │
+└──────────┴──────────┘
+
+Public Key:
+1D********a21F********8WwP********yU/n8vFvM=
+
+$ tree
+.
+├── ed25519.pubkey
+├── ed25519.seckey
+└── output_STDK****ce**2**3
+    └── device_info.json
+```
+Copy the Serial Number and Public Key after running the command on your computer. You will need to upload those values to the SmartThings Cloud via [Developer Workspace](https://smartthings.developer.samsung.com/) during the next phase.   
+
+If you create a device identity with a command with an option like above, You can get the ready-to-use device_info.json file directly.   
```

Comment:
Replace "If you create..." with "After you create..". Otherwise, as a reader I wonder if there are other ways that are not described here. Please ignore this comment if you feel it is not necessary.

Reply:
Has Changed.